src/main/java/com/xwiki/authentication/saml/xwiki/Group.java

```diff
@@ -0,0 +1,112 @@
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.rendering.syntax.Syntax;
+import static java.util.Collections.singletonMap;
+
+public class Group {
+    private static final String XWIKI_GROUP_MEMBERFIELD = "member";
+    private static final EntityReference GROUP_PARENT = new EntityReference("XWikiGroups", EntityType.DOCUMENT,
+            new EntityReference(XWiki.SYSTEM_SPACE, EntityType.SPACE));
+    private final XWikiDocument groupDoc;
+    private final XWikiContext context;
+    private final DocumentReferenceResolver<String> groupResolver;
+
+    public Group(String groupName, XWikiContext context, DocumentReferenceResolver<String> groupResolver) throws XWikiException {
+        this.groupResolver = groupResolver;
+        this.groupDoc = context.getWiki().getDocument(getGroupReferenceForName(groupName), context);
+        this.context = context;
+    }
+
+    public boolean hasMember(String xwikiUserName) throws XWikiException {
+        return getGroupMembers(groupDoc).stream()
+                .anyMatch(groupMember -> Objects.equals(getMemberName(groupMember), xwikiUserName));
+    }
+
+    private String getMemberName(BaseObject groupMember) {
+        return groupMember.getStringValue(XWIKI_GROUP_MEMBERFIELD);
+    }
+
+    private List<BaseObject> getGroupMembers(XWikiDocument groupDoc) throws XWikiException {
+        final List<BaseObject> groupMembers = groupDoc.getXObjects(getGroupClass().getDocumentReference());
+        if (groupMembers == null)
+            return Collections.emptyList();
+
+        return groupMembers.stream().filter(Objects::nonNull).collect(Collectors.toList());
+    }
+
+    public String groupName() {
+        return groupDoc.getDocumentReference().getName();
+    }
+
+    public void addMember(String xwikiUserName) throws XWikiException {
+        addMemberToGroup(xwikiUserName);
+    }
+
+    private void addMemberToGroup(String xwikiUserName) throws XWikiException {
+        BaseObject memberObj = groupDoc.newXObject(getGroupClass().getDocumentReference(), context);
+        getGroupClass().fromMap(singletonMap(XWIKI_GROUP_MEMBERFIELD, xwikiUserName), memberObj);
+    }
+
+    private BaseClass getGroupClass() throws XWikiException {
+        return context.getWiki().getGroupClass(context);
+    }
+
+    public boolean isNew() {
+        return groupDoc.isNew();
+    }
+
+    void setupNewGroupDocument() {
+        groupDoc.setSyntax(Syntax.XWIKI_2_0);
+        groupDoc.setContent("{{include reference='XWiki.XWikiGroupSheet' /}}");
+    }
+
+    public void save() throws XWikiException {
+        context.getWiki().saveDocument(groupDoc, context);
+    }
+
+    public void removeUser(String xwikiUserName) throws XWikiException {
+        getUserMembership(xwikiUserName).map(groupDoc::removeXObject);
```

Comment:
Since you don't use the return value:
```suggestion
        getUserMembership(xwikiUserName).ifPresent(groupDoc::removeXObject);
```